django_graph_api/graphql/utils.py

```diff
@@ -0,0 +1,26 @@
+from graphql.ast import (
+    Field,
+    FragmentSpread,
+    InlineFragment,
+)
+
+
+def get_selections(selections, fragments, object_type):
+    _selections = []
+
+    for selection in selections:
+        if isinstance(selection, Field):
+            _selections.append(selection)
+            continue
+
+        if isinstance(selection, FragmentSpread):
+            fragment = fragments[selection.name]
+        elif isinstance(selection, InlineFragment):
+            fragment = selection
+
+        if fragment.type_condition.name != object_type.object_name:
+            continue
```

Comment:
Should we throw an error here instead of ignoring it?

Reply:
I think that graphql is intended to be forgiving about this. All the spec says is:

> Selections within fragments only return values when concrete type of the object it is operating on matches the type of the fragment.
> - Source: http://facebook.github.io/graphql/October2016/#sec-Type-Conditions

At the very least for unions we might end up with a fragment that doesn't apply to the current object. I think it could make sense to raise an error if the given object type will never be an in the current context, but it doesn't seem like the spec requires it.